.github/workflows/docs.yml

```diff
@@ -0,0 +1,73 @@
+name: Docs
+
+#DEPLOY NOTE: This workflow will only deploy the docs when a workflow is manually triggered from the main branch via github! 
+
+on:
+  pull_request:
+    branches: [ master ] # build docs as additional test on PRs to main
+  workflow_dispatch:
+    branches: [ '*' ] # allow manual trigger for all branches
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+  
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out a copy of the repository
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: sudo apt install graphviz libportaudio2
+      - name: Set up hatch
+        uses: ./.github/actions/setup-hatch
+      - name: Run sphinx
+        run: hatch -v run docs:build
+      - uses: actions/upload-artifact@v4
+        with:
+          name: build-docs
+          path: ./docs/build/html
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: build-docs
+      - name: Upload docs to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v4
+        if: ${{ github.event_name =='workflow_dispatch' && github.ref == 'refs/heads/master' }}
+        with:
+          publish_branch: gh-pages
```

Comment:
I think it would be better to move the `if: ${{ github.event_name =='workflow_dispatch' && github.ref == 'refs/heads/master' }}` statement directly under the `deploy` job statement so that the `deploy` job is skipped (in case it was not triggered manually). Currently, the deploy job is executed, although deployment will never be done since it needs to be triggered manually via the `workflow_dispatch` action.  

If we let it as it is, the job output looks like the docs have been deployed:
![image](https://github.com/acoular/acoular/assets/28733219/2c794ead-05e5-4f79-9d70-de4614c2229e)

But in fact, the deployment (upload to GitHub pages) was skipped: 
![image](https://github.com/acoular/acoular/assets/28733219/50ee8e01-1f42-47d2-b5fa-be863a008bd0)

Reply:
This was intentional to check if the artifacts work. But I see how this could be misleading. It is probably better to skip the enitre job.